map_while_grounded: true

mapping:

  # resolution on the lowest level (fractor = 0, depth = max depth)
  resolution: 0.05 # [m]

  # fractor = how many levels from the lowest level (the higherst res)
  #           should we work on
  # current_resolution = resolution * 2 ^ fractor
  initial_fractor: 3

persistency:

  enabled: false

  map_name: "default"

  # save mapevery "n" seconds, will be save only when flying_normally == true
  save_time: 10.0 # [s]

  # automatically aligns altitude after start
  align_altitude:

    enabled: false

    ground_detection_distance: 2.0 # [m]

    # how high is the fcu fram above the ground when sitting on the ground
    robot_height: 0.3 # [m]

global_map:

  # the publisher rate of the full map
  rate: 1.0 # [Hz]

  # should we prune the map before publishing?
  compress: true

  publish_full: false # should publish map with full probabilities?
  publish_binary: true # should publish map with binary occupancy?

local_map:

  # should create a local map from the global map?
  enabled: true

  # the bounding box distance of the local map
  horizontal_distance: 50.0
  vertical_distance: 50.0

  # the publisher rate of the local map
  rate: 10.0 # [Hz]

  max_computation_duty_cycle: 0.33

  publish_full: false # should publish map with full probabilities?
  publish_binary: true # should publish map with binary occupancy?

unknown_rays:

  update_free_space: true
  clear_occupied: false # beware, setting this to true is very dangerous
  ray_distance: 15.0

# for point cloud raycasting
sensor_params_3d:

  enabled: true
  vertical_fov_angle: deg(33.2)
  vertical_rays: 16
  horizontal_rays: 64 # default 2048

sensor_model:

  max_range: 20.0

  hit: 0.95  # [0, 1]
  miss: 0.45 # [0, 1]
  min: 0.3  # [0, 1]
  max: 0.7  # [0, 1]
